src/huntsman/drp/datatable.py

```diff
@@ -134,36 +162,96 @@ def update_document(self, data_id, data):
             raise ValueError("No matches in database for update.")
         if result.matched_count > 1:
             raise ValueError("Multiple matches in database for update.")
         return result
 
+    @permission_validation
+    def delete_document(self, data_id, **kwargs):
+        """
+        Delete the document associated with the data_id.
+        Args:
+            data_id (dict): Dictionary of key: value pairs identifying the document.
+        Returns:
+            `pymongo.results.UpdateResult`: The result of the delete operation.
+        """
+        self.find(data_id, expected_count=1)  # Make sure there is only one match
+        result = self._table.delete_one(data_id)
+        if result.matched_count == 0:
+            raise ValueError("No matches in database for update.")
+        if result.matched_count > 1:
+            raise ValueError("Multiple matches in database for update.")
+        return result
+
+    def update_file_data(self, filename, data, **kwargs):
+        """
+        Update the metadata associated with a file in the database.
+        Args:
+            filename (str): Modify the metadata for this file.
+            data (dict): Dictionary of key: value pairs to update in the database. The field will
+                be created if it does not already exist.
+        Returns:
+            `pymongo.results.UpdateResult`: The result of the update operation.
+        """
+        data_id = {'filename': filename}
+        return self.update_document(data_id, data, **kwargs)
+
+    def delete_file_data(self, filename, **kwargs):
+        """
+        Delete the metadata associated with a file in the database.
+        Args:
+            filename (str): Modify the metadata for this file.
+            data (dict): Dictionary of key: value pairs to update in the database. The field will
+                be created if it does not already exist.
+        Returns:
+            `pymongo.results.UpdateResult`: The result of the delete operation.
+        """
+        data_id = {'filename': filename}
+        return self.delete_document(data_id, **kwargs)
+
+    def _validate_edit_permission(self, bypass_allow_edits=False, **kwargs):
+        """Raise a PermissionError if not `bypass_allow_edits` or `self._allow_edits`."""
+        if not (bypass_allow_edits or self._allow_edits):
+            raise PermissionError("Edits are not allowed by-default for this table. If you are"
+                                  "sure you want to do this, use `bypass_allow_edits=True`.")
+
+    def _validate_new_document(self, metadata):
+        """Make sure the required columns are in the metadata."""
+        if self._required_columns is None:
+            return
+        missing = [k for k in self._required_columns if k not in metadata.keys()]
+        if len(missing) != 0:
+            raise ValueError(f"Missing columns for update: {missing}.")
+        self.find(metadata, expected_count=0)
+
 
 class RawDataTable(DataTable):
```

Comment:
If this class is huntsman specific-specific, maybe name it so that is clear:

```suggestion
class HuntsmanRawDataTable(DataTable):
```

Reply:
I think the namespace `huntsman.drp.datatable` clears this up already?